packages/core/src/core.ts

```diff
@@ -1429,14 +1429,14 @@ class ThresholdKey implements ITKey {
         privKey: this._localMetadataTransitions[0],
         serviceProvider: this.serviceProvider,
       });
+      this._localMetadataTransitions = [[], []];
+      this.lastFetchedCloudMetadata = this.metadata.clone();
     } catch (error: unknown) {
       throw CoreError.metadataPostFailed(prettyPrintError(error as Error));
+    } finally {
+      // release lock
+      if (acquiredLock) await this.releaseWriteMetadataLock();
     }
-
-    this._localMetadataTransitions = [[], []];
-    this.lastFetchedCloudMetadata = this.metadata.clone();
-    // release lock
-    if (acquiredLock) await this.releaseWriteMetadataLock();
```

Comment:
```
// release lock
 if (acquiredLock) await this.releaseWriteMetadataLock();
 ````

Reply:
This may be implementation specific, though have seen a rethrow in the past escape a finally block if it is not caught within the catch.

Reply:
Then it is fine. You can resolve these to comments and leave it as is. :)